app/src/features/apiClient/helpers/auth.ts

```diff
@@ -3,7 +3,55 @@ import { AUTHORIZATION_TYPES } from "../screens/apiClient/components/clientView/
 import { AUTH_ENTRY_IDENTIFIER } from "../screens/apiClient/components/clientView/components/request/components/AuthorizationView/types";
 import { KeyValuePair, RQAPI } from "../types";
 
-export const processAuthOptions = (authOptions: RQAPI.AuthOptions) => {
+export const processAuthForEntry = (
+  entry: RQAPI.Entry,
+  entryDetails: {
+    id: RQAPI.Record["id"];
+    collectionId: RQAPI.Record["collectionId"];
+  },
+  allRecords: RQAPI.Record[]
+) => {
+  let authOptions = entry.auth;
+  if (entry.auth.currentAuthType === AUTHORIZATION_TYPES.INHERIT) {
+    authOptions = inheritAuth(entry, entryDetails, allRecords);
+  }
+
+  if (!authOptions) {
+    return {
+      headers: [],
+      queryParams: [],
+    };
+  }
+
+  return processAuthOptions(authOptions);
+};
+
+function inheritAuth(
+  entry: RQAPI.Record["data"],
+  entryDetails: {
+    id: RQAPI.Record["id"];
+    collectionId: RQAPI.Record["collectionId"];
+  },
+  allRecords: RQAPI.Record[]
+) {
+  const parentRecord = entryDetails.collectionId
+    ? allRecords.find((record) => record.id === entryDetails.collectionId)
+    : null;
+  if (!parentRecord) {
+    return null;
+  }
+  let parentAuthData = parentRecord.data.auth;
```

Comment:
this can be prevented by consuming collection map once [PR](https://github.com/requestly/requestly/pull/2343) gets merged

Reply:
True and to be clear, this would only optimise line#38 here. 
IMO, I don't feel it would be correct to wait for some other PR to get merged as these changes have already had to be rewritten several times because of conflicting changes